packages/marshal/test/test-marshal-smallcaps.js

```diff
@@ -0,0 +1,219 @@
+// @ts-check
+
+import { test } from './prepare-test-env-ava.js';
+
+import { makeMarshal } from '../src/marshal.js';
+
+import { roundTripPairs } from './test-marshal.js';
+
+const { freeze, isFrozen, create, prototype: objectPrototype } = Object;
+
+// this only includes the tests that do not use liveSlots
+
+const makeTestMarshal = () =>
+  makeMarshal(undefined, undefined, {
+    errorTagging: 'off',
+    useSmallcaps: true,
+  });
+
```

Comment:
I'd like to see a test that serves as a succinct design description of smallcaps so that in less than a page of code we would clearly see:

>  - `$` is slot
>  - `+` or `-` is bigint
> - `#` is constant (e.g., #NaN)
> - `'` is quote so that any of the above can be escaped

and perhaps another showing that smallcaps saves 40% in size in examples such as the one in

https://github.com/Agoric/agoric-sdk/issues/2589#issuecomment-1238686087

Reply:
Between @dtribble and I, the logical tests are now done. I'm gonna resolve this without the size test, but please unresolve if you think I need to do that before merging.